packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -1273,4 +1285,68 @@ export class MirrorNodeClient {
     }
     return result;
   }
+
+  /**
+   * Retrieves and processes transaction record metrics from the mirror node based on the provided transaction ID.
+   *
+   * @param {string} transactionId - The ID of the transaction for which the record is being retrieved.
+   * @param {string} callerName - The name of the caller requesting the transaction record.
+   * @param {string} requestId - The unique identifier for the request, used for logging and tracking.
+   * @param {string} txConstructorName - The name of the transaction constructor associated with the transaction.
+   * @param {string} operatorAccountId - The account ID of the operator, used to calculate transaction fees.
+   * @returns {Promise<{transactionFee: number;} | undefined>} - An object containing the transaction fee if available, or `undefined` if the transaction record is not found.
+   */
+  public async getTransactionRecordMetrics(
+    transactionId: string,
+    callerName: string,
+    requestId: string,
+    txConstructorName: string,
+    operatorAccountId: string,
+  ): Promise<ITransactionRecordMetric | undefined> {
+    const formattedRequestId = formatRequestIdMessage(requestId);
+
+    this.logger.trace(
+      `${formattedRequestId} Get transaction record via mirror node: transactionId=${transactionId}, txConstructorName=${txConstructorName}, callerName=${callerName}`,
+    );
+
+    // poll mirror node to get transaction record
+    const transactionRecords = await this.repeatedRequest(
+      this.getTransactionById.name,
+      [transactionId, 0],
+      this.MIRROR_NODE_GET_CONTRACT_RETRIES,
```

Comment:
QQ: Shouldn't this be `MIRROR_NODE_GET_TRANSACTION_BY_ID_RETRIES`

Reply:
Right great catch! `MIRROR_NODE_GET_CONTRACT_RETRIES` has been used this whole time but you're right it is now used for other purposes rather than just get contract result. I'll come up with a more common name so it can be used for both methods.

Reply:
updated with a more generic name, MIRROR_NODE_REQUEST_RETRY_COUNT

Reply:
Make sure that this env variable is also updated in the Kubernetes secrets for previewnet/testnet/mainnet